### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by arnrdn**
Hello!

In this PR I:
1. Created a class "Nameable"
2. Turned the Person class to Nameable
3. Prepared base Decorator
4. Prepared CapitalizeDecorator and TrimmerDecorator

Thank you for your review!

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Nameable
+  def correct_name(name)
+    corrected_name = name.capitalize
+    corrected_name = name[0...10] if name.length > 10
+    corrected_name
+  end
+end
```
Comment (TedLivist) [inline nameable.rb]: - [ ] According to the instructions, this **Nameable** class should return a descriptive `NotImplementedError`. Kindly fix that.

File: person.rb [hunk 2]
```diff
@@ -1,14 +1,21 @@
+require './nameable'
+
 class Person
```
Comment (TedLivist) [inline person.rb]: - [x] This **`Person`** should inherit from the **`Nameable`** class, like so: 

  ```
  class Person < Nameable
  ```

File: person.rb [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
     @parent_permission = parent_permission
+    @corrector = Nameable.new
   end
 
   attr_accessor :name, :age
   attr_reader :id
 
+  def correct_name
+    @name = @corrector.correct_name(name)
+  end
```
Comment (TedLivist) [inline person.rb]: - [x] The `correct_name` method in this **Person** class should return only `name`

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @rdnrn,

Good job so far!
✔️ There is Nameable class
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] Good job on your README, you are still missing some sections. Kindly format your README using this Microverse template [here](https://github.com/microverseinc/readme-template)

- [ ] There should be a **`Decorator`** class that inherits from the Nameable class with the following features in the image. Kindly implement this.

  rdn-deco

- [ ] You are missing the CapitalizeDecorator and TrimmerDecorator classes with the features in the image below, kindly add them.

  rdn-inhr

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @rdnrn,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**